mongo_connector/doc_managers/nodes_and_relationships_updater.py

```diff
@@ -47,6 +57,10 @@ def update_relationship(self, document, root_type, doc_type, doc_id):
     self.statements_with_params.append(builder.query_nodes)
     self.statements_with_params.append(builder.relationships_query)
 
+  def remove_legacy_nodes(self, doc_id, doc_type):
+    params_dict = {"doc_id": doc_id}
+    statement = "MATCH (d:Document:{doc_type}) WHERE d._id={{doc_id}} OPTIONAL MATCH (d)-[r]-(c) DELETE r, c".format(doc_type=doc_type)
```

Comment:
@hannelita so the root document node is deleted (`d`) with this query? But not the session legacy node? Or am I misunderstanding?